### PR TITLE
Add custom tabular classification model evaluation notebook

### PR DESCRIPTION
**Pull request by krishr2d2**
<br>
This PR makes the following changes to the official folder:

- Adds a notebook for custom-tabular-model-evaluation-pipeline execution in the `official/model_evaluation/` folder.
- Adds an image of the pipeline (displayed in the above notebook) to the `official/model_evaluation/images/` folder.
<br>

**REQUIRED:** Fill out the below checklists or remove if irrelevant
1. If you are opening a PR for `Official Notebooks` under the [notebooks/official](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/official) folder, follow this mandatory checklist:
- [x] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [x] Follow the style and grammar rules outlined in the above notebook template.
- [x] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [x] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [ ] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [x] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [x] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

### CODE REVIEW

File: notebooks/official/model_evaluation/custom_tabular_classification_model_evaluation.ipynb
Comment (andrewferlitsch) [inline notebooks/official/model_evaluation/custom_tabular_classification_model_evaluation.ipynb:1]: <p>Why not just use the scikit-learn prebuilt container?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1219/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='58'/>

Comment (krishr2d2) [reply notebooks/official/model_evaluation/custom_tabular_classification_model_evaluation.ipynb:1]: Tried it but was unable to properly load the sklearn pipeline (preprocessing + estimator) file using the pre-built image. Moreover, the serving application includes the code(postprocessing) to format the predictions as needed.

**Comment by soheilazangeneh** [comment]
This notebook has not been added to the CODEOWNERS while this is checked as done in the PR! Please mark that is done when it's actually added. I uncheck it.

**Comment by krishr2d2** [comment]
Added the notebook to the CODEOWNERS file. Checking the item in the list.

File: notebooks/official/model_evaluation/custom_tabular_classification_model_evaluation.ipynb
Comment (andrewferlitsch) [inline notebooks/official/model_evaluation/custom_tabular_classification_model_evaluation.ipynb:1]: <p>you are going to use in this session. -&gt; you use in this tutorial.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1219/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='38'/>

Comment (andrewferlitsch) [inline notebooks/official/model_evaluation/custom_tabular_classification_model_evaluation.ipynb:1]: <p>only use a few rows(20K) -&gt; only use 20K rows</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1219/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

Comment (andrewferlitsch) [inline notebooks/official/model_evaluation/custom_tabular_classification_model_evaluation.ipynb:1]: <p>as given below -&gt; remove this</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1219/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='50'/>

Comment (andrewferlitsch) [inline notebooks/official/model_evaluation/custom_tabular_classification_model_evaluation.ipynb:1]: <p>add heading</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1219/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='60'/>

Comment (andrewferlitsch) [inline notebooks/official/model_evaluation/custom_tabular_classification_model_evaluation.ipynb:1]: <p>Add heading</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1219/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='62'/>

Comment (krishr2d2) [reply notebooks/official/model_evaluation/custom_tabular_classification_model_evaluation.ipynb:1]: <p>Updated.</p>

Comment (krishr2d2) [reply notebooks/official/model_evaluation/custom_tabular_classification_model_evaluation.ipynb:1]: <p>updated.</p>

Comment (krishr2d2) [reply notebooks/official/model_evaluation/custom_tabular_classification_model_evaluation.ipynb:1]: <p>removed.</p>

Comment (krishr2d2) [reply notebooks/official/model_evaluation/custom_tabular_classification_model_evaluation.ipynb:1]: <p>added.</p>

Comment (krishr2d2) [reply notebooks/official/model_evaluation/custom_tabular_classification_model_evaluation.ipynb:1]: <p>added.</p>

**Comment by krishr2d2** [comment]
> Is this good to go?
> 
> @krishr2d2 if possible, can you replace the setup sections with the new sections in https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb?
> 
> See https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1207/ for the expected changes.

There is no service account permissions section in this updated template. It is needed for this pipeline notebook to run automatically in the CI checks without causing a permission error. Do I need to remove that section too or should I let it be the same in this notebook?

File: notebooks/official/model_evaluation/custom_tabular_classification_model_evaluation.ipynb
Comment (soheilazangeneh) [inline notebooks/official/model_evaluation/custom_tabular_classification_model_evaluation.ipynb:1]: <p>Change to: <strong>Vertex AI Pipelines: Evaluating BatchPrediction results from a Custom Tabular classification model</strong></p><p>For consistency with the other eval notebooks</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1219/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (soheilazangeneh) [inline notebooks/official/model_evaluation/custom_tabular_classification_model_evaluation.ipynb:1]: <p>Add new lines after each constant definition for better readability.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1219/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='39'/>

Comment (soheilazangeneh) [inline notebooks/official/model_evaluation/custom_tabular_classification_model_evaluation.ipynb:1]: <blockquote>Locally, create an application using for serving using&nbsp;<code><a href="https://fastapi.tiangolo.com/tutorial/first-steps/" target="_blank">FastAPI</a></code>&nbsp;Python package.</blockquote><p>using for serving using? I think this needs to be edited :D</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1219/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='54'/>

Comment (soheilazangeneh) [inline notebooks/official/model_evaluation/custom_tabular_classification_model_evaluation.ipynb:1]: <p>Change to:</p><pre class="ql-syntax" spellcheck="false">!gsutil cp "model.joblib" {BUCKET_URI}/{MODEL_ARTIFACT_DIR}/
</pre><p>Added double quotations around model.joblib ow it'll give error that the url was not found!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1219/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='55'/>

Comment (soheilazangeneh) [inline notebooks/official/model_evaluation/custom_tabular_classification_model_evaluation.ipynb:1]: <p>Add new line after each command for better readability.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1219/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='57'/>

Comment (soheilazangeneh) [inline notebooks/official/model_evaluation/custom_tabular_classification_model_evaluation.ipynb:1]: <p>Can we run this on quiet mode? Prints a very long log!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/1219/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='69'/>

Comment (krishr2d2) [reply notebooks/official/model_evaluation/custom_tabular_classification_model_evaluation.ipynb:1]: <p>Yes. Added the --supress-logs flag.</p>

Comment (krishr2d2) [reply notebooks/official/model_evaluation/custom_tabular_classification_model_evaluation.ipynb:1]: <p>added.</p>

Comment (krishr2d2) [reply notebooks/official/model_evaluation/custom_tabular_classification_model_evaluation.ipynb:1]: <p>removed the extra "using".</p>

Comment (krishr2d2) [reply notebooks/official/model_evaluation/custom_tabular_classification_model_evaluation.ipynb:1]: <p>added.</p>

Comment (krishr2d2) [reply notebooks/official/model_evaluation/custom_tabular_classification_model_evaluation.ipynb:1]: <p>updated.</p>

**Comment by krishr2d2** [comment]
It appears that pip cannot install the packages with a --user flag inside the testing environment. It raises the following error: "ERROR: Can not perform a '--user' install. User site-packages are not visible in this virtualenv.". Removing the "--user" flag.

**Comment by soheilazangeneh** [comment]
> s that pip cannot install the packages with a --user flag inside the testing environment. It raises the following error: "ERROR: Can not perform a '--user' install. User site

Where does it raise the error? All the tests have been passed :-curious

**Review by soheilazangeneh — APPROVED** [review]

**Comment by krishr2d2** [comment]
> > s that pip cannot install the packages with a --user flag inside the testing environment. It raises the following error: "ERROR: Can not perform a '--user' install. User site
> 
> Where does it raise the error? All the tests have been passed :-curious

In the first cell where the packages are installed. It was before I removed the --user flag. Checks have been passed without the flag actually.

**Review by soheilazangeneh — CHANGES_REQUESTED** [review]
Blocking the merge to troubleshoot the user flag! @ivanmkc @andrewferlitsch could you please share your insights?

**Comment by ivanmkc** [comment]
@soheilazangeneh Just remove the `--user` flag. We can defer to the user to decide if they want to add one or not on their own system.

**Review by soheilazangeneh — APPROVED** [review]

**Pull request merged**
